### PR TITLE
[improvement](meta) allow to ignore unknown image module

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

Add new FE config `ignore_unknown_metadata_module`. Default is false.
If set to true, when reading metadata image file, and there are unknown modules, these modules
will be ignored and skipped.
This is mainly used in downgrade operation, old version can be compatible with new version Image file.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/persist/meta/MetaReader.java [hunk 1]
```diff
@@ -110,8 +111,17 @@ public static void read(File imageFile, Env env) throws IOException, DdlExceptio
                 }
                 MetaPersistMethod persistMethod = PersistMetaModules.MODULES_MAP.get(metaIndex.name);
                 if (persistMethod == null) {
-                    throw new IOException("Unknown meta module: " + metaIndex.name + ". Known modules: "
-                            + PersistMetaModules.MODULE_NAMES);
+                    if (Config.ignore_unknown_metadata_module) {
+                        LOG.warn("meta modules {} is unknown, ignore and skip it");
```
Comment (cambyzju) [inline fe/fe-core/src/main/java/org/apache/doris/persist/meta/MetaReader.java:115]: forget use module names for '{}' ?

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/persist/meta/MetaReader.java]: fixed

**Comment by morningman** [comment]
run buildall

**Review by Jibing-Li — APPROVED** [review]
LGTM

**Review by AshinGau — APPROVED** [review]

**Comment by AshinGau** [comment]
LGTM

**Review by Kikyou1997 — APPROVED** [review]

**Pull request merged**